packages/ra-core/src/sideEffect/useNotify.ts

```diff
@@ -27,12 +27,14 @@ const useNotify = () => {
             message: string,
             type: NotificationType = 'info',
             messageArgs: any = {},
-            undoable: boolean = false
+            undoable: boolean = false,
+            autoHideDuration: number
```

Comment:
It should probably be optional:

```suggestion
            autoHideDuration?: number
```